sdk/typespec/typespec_client_core/src/http/response.rs

```diff
@@ -13,20 +14,73 @@ pub type PinnedStream = Pin<Box<dyn Stream<Item = crate::Result<Bytes>> + Send +
 #[cfg(target_arch = "wasm32")]
 pub type PinnedStream = Pin<Box<dyn Stream<Item = crate::Result<Bytes>>>>;
 
-/// An HTTP response containing a stream of bytes.
-pub struct RawResponse {
+/// Trait that represents types that can be deserialized from an HTTP response body.
+pub trait Model: Sized {
```

Comment:
I don't think this has to restrict our ability to support "allocation-free" deserialization, presuming what you mean is not 100% allocation free but rather not having to allocate the memory for the entire body. This constraint forces the `Model` type itself to be Sized, but the `LazyBody` test below illustrates a model that just takes the existing body (a Boxed Pinned pointer to an async stream of Bytes) and wraps it up in a "smart-pointer" of sorts. It's not allocation-free, but the `LazyBody` type is essentially the same size as the already-allocated `ResponseBody` that gets moved into it.

In general though, I'm totally on board with enabling allocation-free and low-allocation options. I looked in to supporting `Deserialize` rather than `DeserializeOwned` and it's probably doable, but would need some further investigation to get it working.